### PR TITLE
feat: pagination on the item page for comments + time-ago filter

### PR DESCRIPTION
**Pull request by brunocorrea23**
#212 
#211

### CODE REVIEW

**Comment by iuioiua** [comment]
Are you able to confirm whether this solves #256?

**Comment by brunocorrea23** [comment]
In a way, yes. This PR allows for the user to add more than 10 comments in an item if you maintain PAGE_LENGTH=10.

But the core of the issue is still not solved.
It seems that the issue is connected to the Kv.getMany function on the getManyUsers function. It seems that the Kv.getMany function can't handle an array with more than 10 items as an input. (are you able to confirm if this is a limitation of the getMany function in kv?).

So if you slice the pages in groups of 10 elements (items, comments, etc) or less, you should be fine. But if you change PAGE_LENGTH to 20, for example, it throws an error. This is true not only for the items page, but also for the home page.

**Comment by iuioiua** [comment]
Yes, `PAGE_LENGTH` should not increase beyond 10 because of this current limitation in Deno KV.

File: routes/index.tsx [hunk 1]
```diff
@@ -54,25 +62,12 @@ export const handler: Handlers<HomePageData, State> = {
...
-          max={props.lastPage}
-          value={props.currentPage}
-          // @ts-ignore: this is valid HTML
-          onchange="this.form.submit()"
-        />
-        <label for="current_page" class="whitespace-nowrap align-middle">
-          of {props.lastPage}
-        </label>
-      </form>
+    <div class="flex justify-center">
```
Comment (iuioiua) [inline routes/index.tsx:67]: Please add a note above saying these links do not preserve current URL queries. E.g. if `?page=2`, that'll be removed once one of these links is clicked. This behaviour is fine in this particular context but might need to change in the future as more URL queries control the item list view.

File: routes/item/[id].tsx [hunk 2]
```diff
@@ -12,53 +13,66 @@ import {
...
+  lastPage: number;
+}
+
+function calcPageNum(url: URL) {
+  return parseInt(url.searchParams.get("page") || "1");
+}
+
+function calcLastPage(total = 0, pageLength = PAGE_LENGTH): number {
+  return Math.ceil(total / pageLength);
 }
```
Comment (iuioiua) [inline routes/item/[id].tsx:46]: Can we centralise and de-duplicate any pagination-related code in the utils folder with accompanying tests?

File: routes/item/[id].tsx [hunk 3]
```diff
@@ -12,53 +13,66 @@ import {
...
   getUserBySessionId,
-  getVotedItemsByUser,
   type Item,
   type User,
 } from "@/utils/db.ts";
 import { redirect } from "@/utils/http.ts";
 import UserPostedAt from "@/components/UserPostedAt.tsx";
 import { pluralize } from "@/utils/display.ts";
 
+const PAGE_LENGTH = 10;
```
Comment (iuioiua) [inline routes/item/[id].tsx:29]: Please add a short explainer regarding why this should not exceed 10. Perhaps, there's a GitHub issue in the main https://github.com/denoland/deno repo that could be referenced also.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Sorry for the delay in the review. I've given it a quick read and left some suggestions.

**Comment by iuioiua** [comment]
Hi @brunocorrea23, just checking in. Are you still happy to pursue this PR?

**Comment by brunocorrea23** [comment]
Hey! Yes. Will tackle this today. Couldn't take a look over the week, sorry.

**Review by iuioiua — APPROVED** [review]
Pretty much LGTM! I think it might be best positioned above the item list, though. WDYT?

**Review by iuioiua — APPROVED** [review]
Awesome! LGTM! Thank you, @brunocorrea23.

**Pull request merged**